src/main/java/com/yelp/nrtsearch/server/luceneserver/suggest/CompletionInfixSuggester.java

```diff
@@ -0,0 +1,291 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.suggest;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.PostingsFormat;
+import org.apache.lucene.codecs.lucene84.Lucene84Codec;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.SearcherManager;
+import org.apache.lucene.search.suggest.InputIterator;
+import org.apache.lucene.search.suggest.analyzing.AnalyzingInfixSuggester;
+import org.apache.lucene.search.suggest.document.Completion84PostingsFormat;
+import org.apache.lucene.search.suggest.document.CompletionQuery;
+import org.apache.lucene.search.suggest.document.ContextQuery;
+import org.apache.lucene.search.suggest.document.ContextSuggestField;
+import org.apache.lucene.search.suggest.document.PrefixCompletionQuery;
+import org.apache.lucene.search.suggest.document.SuggestIndexSearcher;
+import org.apache.lucene.search.suggest.document.TopSuggestDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.util.BytesRef;
+
+public class CompletionInfixSuggester extends AnalyzingInfixSuggester {
+
+  private static final String EXACT_TEXT_FIELD_NAME = "text";
+  protected static final String SEARCH_TEXT_FIELD_NAME = "search_text";
+  private static final String PAYLOAD_FIELD_NAME = "payload";
+  private static final boolean DEFAULT_COMMIT_ON_BUILD = false;
+  private static final boolean DEFAULT_SKIP_DUPLICATION = true;
+
+  private final Directory dir;
+
+  public CompletionInfixSuggester(Directory dir, Analyzer analyzer) throws IOException {
+    super(dir, analyzer, analyzer, DEFAULT_MIN_PREFIX_CHARS, DEFAULT_COMMIT_ON_BUILD);
+    this.dir = dir;
+  }
+
+  @Override
+  public List<LookupResult> lookup(
+      CharSequence key, Set<BytesRef> contexts, boolean onlyMorePopular, int num)
+      throws IOException {
+    ensureOpen();
+    IndexSearcher searcher;
+    SearcherManager mgr;
+    List<LookupResult> results;
+
+    synchronized (searcherMgrLock) {
+      mgr = searcherMgr; // acquire & release on same SearcherManager, via local reference
+      searcher = searcherMgr.acquire();
```

Comment:
if we issue `lookup` before issuing `build` this would throw a `NullPointerException` ? If so we could probably throw a more appropriate RuntimeException here instead ?

Reply:
++. Added a check of `searcher manager` at start of `lookup` function. `RuntimeException` will be throw if searchermgr is null.